server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -433,6 +434,8 @@ public ClusterState execute(ClusterState currentState) {
 
                             if (entry.state() == State.ABORTED) {
                                 entries.add(entry);
+                                assert entry.shards().isEmpty();
+                                hadAbortedInitializations = true;
```

Comment:
Since we now don't finalise this case in `applyClusterState` we  need to end it when the cluster state for the init has been processed, so we set the flag here to mark that case.